### DISCUSSION TITLE
Issues with Spotify / No sound

### DISCUSSION THREAD

**Discussion by schtzngrm** [rern/rAudio#762] [General]
Hi,

since a while , I'm trying to setup Spotify @ rAudio.  rAudio is running on an Raspberry Pi 2B Rev 1.1 with PiZeroAudio card.
However, so far no success in getting any sound output when using Spotify. Internetradio with streaming url works perfectly!

Today, I did a clean new install, setting up everything from scratch. rAudio shows the selected Spotify music title and progress, but no sound... I'm using a PiZeroAudio card, tried with "Hifiberry DAC+ lite" and "Pimoroni pHAT DAC" as drivers.

I included below the information on Spotifyd. Hope this helps, as I lack experience in programming...

Anything I'm missing?

Best regards
Gerald

BTW: I love rAudio! Using it daily as Internetradio. Thanks for the great work!

________________________________________-

spotifyd 0.3.3-1
[global]
backend = "alsa"
device = "default:CARD=sndrpirpidac"
mixer = "( not available )"
control = "hw:0"
volume_controller = "alsa"
bitrate = 320
onevent = "/srv/http/bash/spotifyd.sh"
use_mpris = false

● spotifyd.service - A spotify playing daemon
     Loaded: loaded (/etc/systemd/system/spotifyd.service; enabled; vendor preset: disabled)
     Active: active (running) since Sun 2022-01-16 11:55:08 CET; 7min ago
    Process: 2160 ExecStartPost=/srv/http/bash/cmd.sh nicespotify (code=exited, status=0/SUCCESS)
   Main PID: 2159 (spotifyd)
      Tasks: 2 (limit: 2133)
        CPU: 477ms
     CGroup: /system.slice/spotifyd.service
             └─2159 /usr/bin/spotifyd --no-daemon

Jan 16 11:57:53 rAudio taskset[2159]: couldn't parse packet from 192.168.178.44:5353: type 47 is invalid
Jan 16 11:58:22 rAudio taskset[2159]: couldn't parse packet from 192.168.178.44:5353: type 47 is invalid
...

**Comment by rern** [comment]
- Show me:
```sh
aplay -L
```
- Try:
```sh
sed -i '/^mixer\|^control\|^volume/ d' /etc/spotifyd.conf
systemctl restart spotifyd
```

**Comment by schtzngrm** [comment]
Dear rern,

proposed solution works! Thank you for the quick reply!

One more thing: Currently, volume changes in rAudio when playing Spotify do nothing, volume can only be changed on the phone used to stream Spotify to rAudio. 

Any idea on how I can set the volume also in rAudio?

Best regards
Gerald

BTW:  aplay -L results for reference, if needed

rAudio:~  #  aplay -L
null
    Discard all samples (playback) or generate zero samples (capture)
pipewire
    PipeWire Sound Server
default:CARD=sndrpirpidac
    snd_rpi_rpi_dac, RPi-DAC HiFi pcm1794a-codec-0
    Default Audio Device
sysdefault:CARD=sndrpirpidac
    snd_rpi_rpi_dac, RPi-DAC HiFi pcm1794a-codec-0
    Default Audio Device

**Reply by rern** [reply]
Hi Gerald,

Nothing we can do if there's no hardware mixers.
To verify if there's actually a hardware mixer on your PiZeroAudio card, show me:
```sh
amixer
```

**Comment by schtzngrm** [comment]
Hi rern,

no result on amixer. I guess the card has no hardware mixer then, only software.
Thank you for the great support! 

Best regards

Gerald
